api/src/main/java/org/openmrs/module/registrationcore/api/impl/PatientEditListener.java

```diff
@@ -0,0 +1,20 @@
+package org.openmrs.module.registrationcore.api.impl;
+
+import org.openmrs.Patient;
+import org.openmrs.event.Event;
+import org.openmrs.module.registrationcore.RegistrationCoreConstants;
+
+import javax.jms.Message;
+
+public class PatientEditListener extends PatientActionListener {
```

Comment:
@djazayeri yep, this one is much smaller. The reason is that:
When "edit patient" event was handled - this class perform:
1. Delegate to "Updater patient" class to perform update on MPI server.

But when "created patient" event was handled - PatientCreationListener perform:
1. Delegate to "Export patient" class to perform export patient to MPI server.
2. Get generated Identifier from MPI server.
3. Add new one patient identifier to local patient and perform update it. (And sure at this moment we need a lot of logic related to granting/resetting privileges with try/finally clauses).